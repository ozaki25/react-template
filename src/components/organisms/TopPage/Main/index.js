import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import BasicText from 'src/components/atoms/texts/BasicText';
import LinkText from 'src/components/atoms/texts/LinkText';
import MainMenuButton from 'src/components/molecules/MainMenuButton';

function Main({ onClickStartReissue }) {
  return (
    <Container>
      <MainMenuContainer>
        <MainMenuButton
          title="利用できないキャッシュカードの再発行"
          description="磁気不良、ICチップ不良、汚損、破損等で利用できなくなったキャッシュカードの再発行"
          onClick={onClickStartReissue}
        />
      </MainMenuContainer>
      <BasicText>※今後、本アプリでできる手続を順次拡大していく予定です。</BasicText>
      <LinkContainer>
        <LinkText>ライセンスについて</LinkText>
      </LinkContainer>
    </Container>
  );
}

export const Container = styled.section`
  padding: 15px 10px 20px;
`;

export const MainMenuContainer = styled.div`
  text-align: center;
`;

export const LinkContainer = styled.div`
  text-align: right;
`;

Main.propTypes = {
  onClickStartReissue: PropTypes.func.isRequired,
};

export default Main;
