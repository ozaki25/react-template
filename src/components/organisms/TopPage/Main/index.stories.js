import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';

import Main from '.';

const stories = storiesOf('organisms/TopPage', module);

stories.add(
  'Main',
  withInfo('トップページのメインコンテンツ')(() => (
    <Main onClickStartReissue={action('click start reissue')} />
  )),
);
