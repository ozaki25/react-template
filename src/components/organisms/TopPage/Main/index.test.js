import React from 'react';
import { snapshot } from 'test/helpers';
import Main, { Container, MainMenuContainer, LinkContainer } from '.';

const props = {
  onClickStartReissue: jest.fn(),
};

snapshot('Main', <Main {...props} />);
snapshot('Main/Container', <Container />);
snapshot('Main/StartReissueButtonContainer', <MainMenuContainer />);
snapshot('Main/LinkContainer', <LinkContainer />);
