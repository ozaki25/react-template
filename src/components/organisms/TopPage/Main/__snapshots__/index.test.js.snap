// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: Main 1`] = `
<styled.section>
  <styled.div>
    <MainMenuButton
      description="磁気不良、ICチップ不良、汚損、破損等で利用できなくなったキャッシュカードの再発行"
      onClick={[Function]}
      title="利用できないキャッシュカードの再発行"
    />
  </styled.div>
  <BasicText>
    ※今後、本アプリでできる手続を順次拡大していく予定です。
  </BasicText>
  <styled.div>
    <LinkText
      onClick={null}
    >
      ライセンスについて
    </LinkText>
  </styled.div>
</styled.section>
`;

exports[`should match snapshot: Main/Container 1`] = `
.c0 {
  padding: 15px 10px 20px;
}

<section
  className="c0"
/>
`;

exports[`should match snapshot: Main/LinkContainer 1`] = `
.c0 {
  text-align: right;
}

<div
  className="c0"
/>
`;

exports[`should match snapshot: Main/StartReissueButtonContainer 1`] = `
.c0 {
  text-align: center;
}

<div
  className="c0"
/>
`;
